src/Statistics.jl

```diff
@@ -103,6 +103,16 @@ julia> mean(√, [1 2 3; 4 5 6], dims=2)
 """
 mean(f, A::AbstractArray; dims=:) = _mean(f, A, dims)
 
+function mean(f::Number, itr::Number)
+    f_value = try
+        f(itr)
+    catch MethodError
+        rethrow(ArgumentError("""mean(f, itr) requires a function and an iterable.
+                                 Perhaps you meant middle(x, y)?""",))
```

Comment:
How about not putting a newline in the error message?
```suggestion
        rethrow(ArgumentError("mean(f, itr) requires a function and an iterable. " *
                              "Perhaps you meant middle(x, y)?"))
```

Reply:
I was following the pattern of
```
julia> (4+1)(5)
ERROR: MethodError: objects of type Int64 are not callable
Maybe you forgot to use an operator such as *, ^, %, / etc. ?
Stacktrace:
 [1] top-level scope
   @ REPL[9421]:1
```
Putting the hint on a different line as the error, but either way works.

Reply:
Ah OK. I'm not sure whether there are strong conventions in that regard.